cdt-cpp man page added (#139)
AntelopeIO/cdt: docs/man/cdt-cpp.1.md

Reviewer: There is a consistency issue in the docs: some sentences end in a period (`.`) and some do not. See `-C` and `-U=<name>` for an example of the inconsistency.

This isn't a deal killer - we don't NEED to make a change here - but it would be nice if we had consistency.

Author: > There is a consistency issue in the docs: some sentences end in a period (`.`) and some do not. See `-C` and `-U=<name>` for an example of the inconsistency.
> 
> This isn't a deal killer - we don't NEED to make a change here - but it would be nice if we had consistency.

I agree but read https://github.com/AntelopeIO/cdt/pull/139#discussion_r1165400310

---
title: CDT-CPP
section: 1
header: User Manual

> footer: cdt-cpp

Reviewer: I think it's more consistent with other projects to use our organization here.
```suggestion
footer: AntelopeIO
```

date: April 08, 2023
---
# NAME

> cdt-cpp - Eosio C++ to WebAssembly compiler

Reviewer: I *think* this would be better?
```suggestion
cdt-cpp - Antelope smart contract C++ to WebAssembly compiler
```

# SYNOPSIS

`cdt-cpp [options] <input file>`

# DESCRIPTION

> **cdt-cpp**  Is a C++ compiler which converts C++ code of the smart contracts to WebAssemply binary code for 
> execution in a virtual machine of Antelope block chains  

Reviewer: ```suggestion
**cdt-cpp**  Is a C++ compiler which converts C++ smart contract code to WebAssemply for 
execution in Antelope block chain virtual machines  
```

# OPTIONS

**`-C`** 

    Include comments in preprocessed output
    
**`--CC`** 

    Include comments from within macros in preprocessed output
    
**`-D=<name definition>`** 

    Define <macro> to <value> (or 1 if <value> omitted)

**`-U=<name>`** 

>     Cancel any previous definition of name, either built in or provided with a -D option.              

Reviewer: Yeah, but no. Feel free to leave it as is.

    
**`-E`**

    Only run the preprocessor    
    
**`-I=<dir>`**
    
    Add directory to include search path
    
**`-L=<dir>`**

    Add directory to library search path    
    
**`--MF=<file>`**

    Write depfile output    

**`--MD`**

    Write depfile containing user and system headers    
    
**`--MMD`**
    
    Like -MD except mention only user header files, not system header files. 
    
**`--MT=<target>`**

    Specify name of main file output in depfile    
    
**`-O=<optimization_level>`**
    

>     Optimization level s, 0-3     

Reviewer: I concur, it is from `cdt-spp`'s help. I think the help is erroneous. I suspect this should read:
`-Os` or even `-O=<s>` instead of `-O=<string>`. 
I think it's kind of unclear as written. I suppose I expect to write `-O2` etc.

>     Optimization level s, 0-3     

Author: > I concur, it is from `cdt-spp`'s help. I think the help is erroneous. I suspect this should read: `-Os` or even `-O=<s>` instead of `-O=<string>`. I think it's kind of unclear as written. I suppose I expect to write `-O2` etc.

@ScottBailey I can say even more: most of description in the --help text are useless or wrong, or the options don't work at all. It was the cause why I decided to stop writing of beautiful descriptions and make the man page with descriptions from --help. If I will test every option, research how it really works by its source code and then write its description it takes a lot of time. Instead of this I will fast create man pages with the descriptions from --help text, it allows to add them to building process, to installation packages, to documentation (that we have man pages) and anywhere they are needed as entities. And then I will update their content when it will be good for other tasks. In short we don't need to make the descriptions beautiful right now. I already have beautiful descriptions for many of these options I just rolled them back because too many options need to be tested and fixed.

>     Optimization level s, 0-3     

Reviewer: @jolly-fellow Did you write an issue to update the help? If not, let's write one that addresses our concerns so we can move on.

    
**`-R=<dir>`**
    
    Add the directory dir to the list of directories to be searched for resource files.
    
**`-S`**

    Only run preprocess and compilation steps    
    
**`-W=<warning_name>`**
    
    Enable the specified warning.   
    
**`--abi-version=<string>`**
    
    Which ABI version to generate    
    
**`--abigen`**
    
    Generate ABI
    
**`--abigen_output=<string>`**
    
    ABIGEN output
    
**`--allow-names`**
    

>     Allows name section to be created

Reviewer: ```suggestion
    Allow creation of name section
```

    
**`--dI`**
    
    Print include directives in -E mode in addition to normal output
    
**`--dM`**
    
    Print macro definitions in -E mode instead to normal output
    
**`--disable-symbolication`**
    
    Disable symbolizing crash backtraces.
    
**`--emit-ast`**
    
    Emit Clang AST files for source inputs
    
**`--emit-ir`**
    
    Emit llvm ir
    
**`--emit-llvm`**
    
    Use the LLVM representation for assembler and object files
    
**`--eosio-pp-dir=<string>`**
    
    Set the directory for eosio-pp
    
**`--fPIC`**
    
    Generate position independent code. This option is used for shared libraries
    
**`--faligned-allocation`**
    
    Enable C++17 aligned allocation functions
    
**`--fasm`**
    
    Assemble file for x86-64
    
**`--fcolor-diagnostics`**
    
    Use colors in diagnostics
    
**`--fcoroutine-ts`**
    
    Enable support for the C++ Coroutines TS
    
**`--finline-functions`**
    
    Inline suitable functions
    
**`--finline-hint-functions`**
    
    Inline functions which are (explicitly or implicitly) marked inline
    
**`--fmerge-all-constants`**
    
    Allow merging of constants
    
**`--fnative`**
    

>     Compile and link for x86-64

Reviewer: Is this true on an M1/M2 Mac?

Author: > Is this true on an M1/M2 Mac?

I don't know. Read my explanation here: https://github.com/AntelopeIO/cdt/pull/139#discussion_r1165400310
